File: LeetCode_Solutions/1256-rank-transform-of-an-array/rank-transform-of-an-array.java
class Solution {
    public int[] arrayRankTransform(int[] arr) {
        if(arr.length<1){
            return arr;
        }
        int count =1;
        int[] arr1=Arrays.copyOf(arr,arr.length);
        Arrays.sort(arr1);
       Map<Integer,Integer>mp=new HashMap<>();
       mp.put(arr1[0],1);
        for(int i=1;i<arr1.length;i++){
            if(mp.containsKey(arr1[i])){
                continue;
            }
            else{
                count=count+1;
                mp.put(arr1[i],count);
            }
        }
      //  System.out.println(mp);
        for(int i=0;i<arr1.length;i++){
                arr[i]=mp.get(arr[i]);
        }
        return arr;
    }
}

File: LeetCode_Solutions/1256-rank-transform-of-an-array/rank-transform-of-an-array.py
class Solution:
    def arrayRankTransform(self, arr: List[int]) -> List[int]:
        if len(arr)<1:
            return arr
        arr1=sorted(arr)
        map={}
        map[arr1[0]]=1
        cnt=1
        for i in range(1,len(arr1)):
            if arr1[i] not in map:
                cnt+=1
                map[arr1[i]]=cnt
        return [map[i] for i in arr]

